internal/scaffold/go_mod_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaffold
```

Comment:
Nit, question, comment :)

Go allows `_test.go` files to be in a different package within the same directory (e.g. `package scaffold_test`).

The reason for doing this is to force the test to use only the public API of the package, which can be helpful to remind us what our public API is and to nudge us towards better public APIs in general. Not sure how prevalently we're doing this elsewhere in the project, but since this PR is _only_ unit tests, you have my attention about unit testing techniques 😉. 

It looks like the test uses only exported types and fields from `package scaffold`, so I'm curious what you think of moving this to `package scaffold_test`, importing `scaffold`, and making this a black box test?

Totally non-blocking.

Reply:
Except we haven't done that yet :) 
```
[jesusr@transam scaffold{add_tests}]$ git grep package *_test.go | grep scaffold
add_controller_test.go:package scaffold
addtoscheme_test.go:package scaffold
apis_test.go:package scaffold
boilerplate_go_txt_test.go:package scaffold
build_dockerfile_test.go:package scaffold
cmd_test.go:package scaffold
controller_kind_test.go:package scaffold
controller_test.go:package scaffold
cr_test.go:package scaffold
doc_test.go:package scaffold
entrypoint_test.go:package scaffold
gitignore_test.go:package scaffold
go_mod_test.go:package scaffold
group_test.go:package scaffold
operator_test.go:package scaffold
register_test.go:package scaffold
role_test.go:package scaffold
rolebinding_test.go:package scaffold
service_account_test.go:package scaffold
tools_test.go:package scaffold
types_test.go:package scaffold
usersetup_test.go:package scaffold
version_test.go:package scaffold
```
For unit tests, I preferred being able to access some of the private bits, mostly to create the negative testing which I've found really hard to do when only using the public interface. In Java, we used to do that and stopped because again sometimes that caused the test setup to get extremely complicated and prevented folks from adding tests.

Reply:
@joelanford forgot to mention you in my reply ^^^

Reply:
I think like @jmrodri. 

IMO is better keep it impl with your own *_test.go file as done here. It makes easier we identify and to keep the tests maintained. Also, I understand that it is following the good practices of the language as well. And then, if we centralize them in a big file it would make very harder identify what is or not covered and etc...